### DISCUSSION TITLE
Trend Finder with Coefficient of Variation

### DISCUSSION THREAD

**Discussion by ShadyNagy** [DaveSkender/Stock.Indicators#1023] [Ideas]
#### Trend Finder with Coefficient of Variation 
Is this indicator exists or there is any way to implement it?
I found it in the trading view on this link with the script https://www.tradingview.com/script/XseYRWFH-Trend-Finder-with-Coefficient-of-Variation/

I will be thankful if this we can implement.

[Update]
One more question:
Did we have `median` function somewhere in the package?
```csharp
bool is_ranging = COV < ta.median(COV, length_m)
```

**Comment by DaveSkender** [comment]
While I don't think the full indicator would pass [our sniff test](https://github.com/DaveSkender/Stock.Indicators/discussions/1024) for direct inclusion in the library, it might make for a good [custom indicator](https://dotnet.stockindicators.dev/custom-indicators/#content).  Our [standard deviation](https://dotnet.stockindicators.dev/indicators/StdDev/#stddevresult) indicator returns the information you'd need.

If I understood this right, it’s simply:

```csharp
// base COR indicator
var results = quotes
  .GetStdDev(..)
  .Select(x => x.Date, x.StdDev / x.Mean);
```

We're also planning to add more [array math functions](https://github.com/DaveSkender/Stock.Indicators/issues/979) for things like median.

**Comment by ShadyNagy** [comment]
@DaveSkender 
Thank you so much maybe I will make PR for array math functions which I am testing now.
Is there a function to convert the data to candles by period?
```csharp
public static List<TQuote> ToCandles<TQuote>(this IEnumerable<(DateTime Date, double Value)> tuples, int candlePeriod)
```

**Reply by DaveSkender** [reply]
Thank you, we love contributions.  

Regarding the candle converter, you’d need all elements to properly form the shape, open, high, low, close prices. I may be misunderstanding.  If you’re seeking to convert bar time size, we have a [.Aggregate(..)](https://dotnet.stockindicators.dev/utilities/#resize-quote-history) utility.
